.gitlab/functional_test/regression_detector.yml

```diff
@@ -91,27 +100,53 @@ single-machine-performance-regression_detector:
+    - echo "Tags passed through SMP are ${SMP_TAGS}"
     - RUST_LOG="info,aws_config::profile::credentials=error"
     - RUST_LOG_DEBUG="debug,aws_config::profile::credentials=error"
-    - RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE}
-      job submit
-      --baseline-image ${BASELINE_IMAGE}
-      --comparison-image ${COMPARISON_IMAGE}
-      --baseline-sha ${BASELINE_SHA}
-      --comparison-sha ${CI_COMMIT_SHA}
-      --target-config-dir test/regression/
-      --submission-metadata submission_metadata
+    - |
+      RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE} \
+      job submit \
+      --baseline-image ${BASELINE_IMAGE} \
+      --comparison-image ${COMPARISON_IMAGE} \
+      --baseline-sha ${BASELINE_SHA} \
+      --comparison-sha ${CI_COMMIT_SHA} \
+      --target-config-dir test/regression/ \
+      --submission-metadata submission_metadata \
+      --tags ${SMP_TAGS} || {
+        exit_code=$?
+        echo "job submit command failed"
+        datadog-ci tag --level job --tags smp_failure_mode:"job-submit"
+        exit $exit_code
+      }
+    # Get the SMP Job Id from 'submission-metadata' and use 'datadog-ci'
+    # to tag the gitlab job seen in CI Visibility with the SMP Job Id.
+    - SMP_JOB_ID=$(jq -r '.jobId' submission_metadata)
+    - echo "SMP Job Id is ${SMP_JOB_ID}"
+    - datadog-ci tag --level job --tags smp_job_id:${SMP_JOB_ID}
     # Wait for job to complete.
-    - RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE}
-      job status
-      --wait
-      --wait-delay-seconds 60
-      --submission-metadata submission_metadata
+    - |
+      RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE} \
+      job status \
+      --wait \
+      --wait-delay-seconds 60 \
+      --submission-metadata submission_metadata || {
+        exit_code=$?
+        echo "job status command failed"
+        datadog-ci tag --level job --tags smp_failure_mode:"job-status"
+        exit $exit_code
+      }
     # Now that the job is completed pull the analysis report, output it to stdout.
-    - RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE}
-      job sync
-      --submission-metadata submission_metadata
-      --output-path outputs
+    - |
+      RUST_LOG="${RUST_LOG}" ./smp --team-id ${SMP_AGENT_TEAM_ID} --api-base ${SMP_API} --aws-named-profile ${AWS_NAMED_PROFILE} \
+      job sync \
+      --submission-metadata submission_metadata \
+      --output-path outputs || {
+        exit_code=$?
+        echo "job sync command failed"
+        datadog-ci tag --level job --tags smp_failure_mode:"job-sync"
+        exit $exit_code
```

Comment:
Is this a behavior change? If `job sync` failed with a non-zero code prior to this change, does that exit the job?

Reply:
From my understanding the prior behavior was to exit since errors in job sync bubble all the way up to main which would panic. I have not seen an example of job sync specifically failing but when any other command fails the script exits so I assume the same here.